lib/commons/odata/ODataClient.js

```diff
@@ -166,7 +186,26 @@ module.exports = class ODataClient {
     createdRecord.isNew = true;
     this.emitter.emit('data', messages.newMessageWithBody(createdRecord));
   }
+  async deleteObjectByField(msg) {
+    const objectType = this.cfg.objectType;
+    const fieldValue = msg.body.FieldsValue;
+    console.log(`Will perform delete to ${objectType} with field ${fieldValue}`);
+ 
+    const response = await this.restClient.makeRequest({
+    url: `${objectType}(${"'"+fieldValue+"'"})`,
```

Comment:
There is one difficult edge case here: If `fieldValue` is wrapped in quotes, this statement will not work when the field is a  GUID or number.